acvm/src/compiler/transformers/fallback.rs

```diff
@@ -6,47 +6,50 @@ use acir::{
 };
 
 // A predicate that returns true if the black box function is supported
-pub type IsBlackBoxSupported = fn(&BlackBoxFunc) -> bool;
+pub type IsOpcodeSupported = fn(&Opcode) -> bool;
 
 pub struct FallbackTransformer;
 
 impl FallbackTransformer {
     //ACIR pass which replace unsupported opcodes using arithmetic fallback
     pub fn transform(
         acir: Circuit,
-        is_supported: IsBlackBoxSupported,
+        is_supported: IsOpcodeSupported,
     ) -> Result<Circuit, CompileError> {
         let mut acir_supported_opcodes = Vec::with_capacity(acir.opcodes.len());
 
         let mut witness_idx = acir.current_witness_index + 1;
 
         for opcode in acir.opcodes {
-            let bb_func_call = match &opcode {
-                Opcode::Arithmetic(_) | Opcode::Directive(_) | Opcode::Block(_, _) => {
-                    // directive, arithmetic expression or block are handled by acvm
+            match &opcode {
+                Opcode::Arithmetic(_)
+                | Opcode::Directive(_)
+                | Opcode::Block(_)
```

Comment:
Whats the difference between a directive and Block/RAM/ROM?

Reply:
If the opcode is supported by the backend, the backend will layout the constraints for the memory block
If the opcode is not supported by the backend, noir will layout the constraints
In both cases, the solver needs to resolve the witness used in the memory trace.